astropy/table/tests/test_pprint.py

```diff
@@ -528,13 +529,24 @@ def test_pprint_npfloat32():
 
 def test_pprint_py3_bytes():
     """
-    Test for #1346.  Make sure a bytestring (dtype=S<N>) in Python 3 is printed
-    correctly (without the "b" prefix like b'string').
+    Test for #1346 and #4944. Make sure a bytestring (dtype=S<N>) in Python 3
+    is printed correctly (without the "b" prefix like b'string').
+    Also make sure special characters are printed in Python 2.
     """
-    val = bytes('val', encoding='utf-8') if PY3 else 'val'
-    dat = np.array([(val,)], dtype=[(str('col'), 'S3')])
+    val = 'val' if PY2 else bytes('val', encoding='utf-8')
```

Comment:
Can you get rid of one or both of the xfails below by setting `val = str('val')`?  In this case it really should work always, no?